src/metabase/public_settings.clj

```diff
@@ -257,12 +258,29 @@
   :visibility :public
   :audit      :getter)
 
+(defn- coerce-to-relative-url
+  "Get the path of a given URL if the URL contains an origin.
+   Otherwise make the landing-page a relative path."
+  [landing-page]
+  (cond
+    (u/url? landing-page) (-> landing-page io/as-url .getPath)
+    (empty? landing-page) ""
+    (not (str/starts-with? landing-page "/")) (str "/" landing-page)
+    :else landing-page))
+
 (defsetting landing-page
   (deferred-tru "Default page to show people when they log in.")
   :visibility :public
   :type       :string
   :default    ""
-  :audit      :getter)
+  :audit      :getter
+  :setter     (fn [new-landing-page]
+                (when new-landing-page
+                  ;; If the landing page is a valid URL or mailto, sms, or file, then check with if site-url has the same origin.
+                  (when (and (or (re-matches #"^(mailto|sms|file):(.*)" new-landing-page) (u/url? new-landing-page))
```

Comment:
is there a reason now to throw for any non-HTTP/HTTPS URLs?

Reply:
hm good point, I think `io/as-url` won't work without the protocol.